text/0000-pack-publish-buildpack.md

```diff
@@ -0,0 +1,276 @@
+
+The implementation of adding and yanking buildpacks will require new configuration elements in Pack's config file, and two new commands in Pack.
+
+## Configuration
+
+A default registry will be configured in `.pack/config.toml` with the following schema.
+
+##### Base Schema
+```toml
+default-registry="<string>"
+
+[[registries]]
+name="<string>"
+type="<string>"
+# <additional per "type" config entries>
+```
+
+The `default-registry` value must match one of the tables in the `[[registries]]` array with the same `name`. The `name` value can be any string.
+
+The `type` key in the `[[registries]]` tabl can be set to one of two values, described below:
+
+##### Type: GitHub Schema
+```toml
+[[registries]]
+name="<string>"
+type="github"
+url="<url to git repo>"
+issues-url="<url to issues for publishing>" # optional, default: "<url>/issues"
+```
+
+##### Type: Git Schema
+```toml
+[[registries]]
+name="<string>"
+type="git"
+url="<path to git repo>"
+```
+
+#### Default
+
+The following will be the implied default when no registries are present, however, it can be overridden by defining a registry with the name "official".
+
+```toml
+default-registry="official"
+
+[[registries]]
+name="official"
+type="github"
+url="https://github.com/buildpack/registry"
+issues-url="https://github.com/buildpack/registry/issues"
+```
+
+## Format
+
+### Type: Git
+
+A commit would be created with the following message:
+
+```
+[<ADD|YANK>] <namespace>/<name>@<version>
+
+<user message (optional)>
+```
+
+##### Example
+
+```
+[YANK] buildpacks-io-samples/hello-universe@0.0.1
+
+Major bug issues found. See https://github.com/buildpacks/samples/issues.
+```
+
+### Type: GitHub issue
+
+An issue would be created with the following content:
+
+- Title: `[<ADD|YANK>] <namespace>/<name>@<version>`
+- Body:
+
+        <user message (optional)>
+
+        ### Data
+
+        ```toml
+        id = <ns/name (required)>
+        version = <version (required)>
+        addr = <address-to-image (only required for ADD)>
+        digest = <digest-of-image (only required for ADD)>
+        ```
+
+##### Example
+
+- Title: `[YANK] samples/hello-universe@0.0.1`
+- Body:
+
+        Major bug issues found. See https://github.com/buildpacks/samples/issues.
+
+        ### Data
+
+        ```toml
+        id = samples/hello-universe
+        version = 0.0.1
+        ```
+
+## `pack publish-buildpack <url>`
+
+Arguments:
+
+* `url` - the location of a buildpackage _in a Docker registry_ to be released in the buildpack registry
+
+Steps:
+
+The following behavior will execute when the configured registry is of type `github`.
+
+1. Fetches the image config blob for the buildpackage at `url`.
+1. Reads the following information from the image config (if any of these are missing, the command fails):
+    - `id` (must include a `namespace` and `name` in the form `<namespace>/<name>`)
+    - `version`
+1. Opens a link to a new Github Issue in the user's browser
```

Comment:
The original RFC suggested opening a PR with the changes in them. Since we already need to code this into `pack` for generating the JSON payload, does it make sense to do this for the GitHub pathway as well?